modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/rel/ScanNode.java

```diff
@@ -131,13 +132,15 @@ private void push() throws Exception {
                     return;
                 }
             }
+        } catch (QueryCancelledException e) {
+            throw new QueryCancelledException(e);
```

Comment:
```suggestion
        } catch ((QueryCancelledException | SqlException e) {
            throw e;
```
Do you see some issue in such approach?